src/main/java/zero/util/Storage.java

```diff
@@ -0,0 +1,93 @@
+package zero.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import zero.exception.ZeroException;
+import zero.task.Deadline;
+import zero.task.Event;
+import zero.task.Task;
+import zero.task.TaskList;
+import zero.task.Todo;
+
+/**
+ * The {@code Storage} class handles the loading and saving of tasks to a file.
+ * It reads tasks from a specified file during initialisation and writes tasks back to the file when required.
+ */
+public class Storage {
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the specified file and returns them as an {@code ArrayList<Task>}.
+     * If the file does not exist, it creates the file and any necessary directories.
+     *
+     * @return An {@code ArrayList<Task>} containing the loaded tasks.
+     * @throws ZeroException If an error occurs while loading tasks.
+     */
```

Comment:
I really like how descriptive and in-depth your Javadoc comments are! This makes it very simple for a fellow developer/reviewer to understand the code you are writing. Great work!